src/units/unit.cpp

```diff
@@ -1814,7 +1811,28 @@ int unit::resistance_against(const std::string& damage_name,bool attacker,const
 			res = 100 - resist_effect.get_composite_value();
 		}
 	}
+	return res;
+}
 
+int unit::resistance_against(const std::string& damage_name,bool attacker,const map_location& loc, const_attack_ptr weapon, const_attack_ptr opp_weapon) const
+{
+	std::pair<std::string, std::string> types;
+	if(opp_weapon){
+		types = opp_weapon->damage_type();
+	} else{
+		types.first = damage_name;
+	}
+	bool has_second_type = !(types.second).empty();
+	int res = movement_type_.resistance_against(types.first);
+	int res_sec = has_second_type ? movement_type_.resistance_against(types.second) : 0;
```

Comment:
`res_sec` is only used within the `if(has_second_type)` block, so its declaration can be moved into that block.